server/src/main/java/org/opensearch/gateway/ClusterStateUpdaters.java

```diff
@@ -120,7 +121,19 @@ static ClusterState updateRoutingTable(final ClusterState state) {
         // initialize all index routing tables as empty
         final RoutingTable.Builder routingTableBuilder = RoutingTable.builder(state.routingTable());
         for (final IndexMetadata cursor : state.metadata().indices().values()) {
-            routingTableBuilder.addAsRecovery(cursor);
+            // Whether IndexMetadata is recovered from local disk or remote it doesn't matter to us at this point.
+            // We are only concerted about index data recovery here. Which is why we only check for remote store enabled and not for remote
+            // cluster state enabled.
+            if (cursor.getSettings().getAsBoolean(IndexMetadata.SETTING_REMOTE_STORE_ENABLED, false) == false
+                || state.routingTable()
```

Comment:
The second check is for remote store indices where remote cluster state is disabled, right?

Reply:
second check is to make sure we have gone through the remote store restore flow and the routing tables have recovery source are remote store.

> The second check is for remote store indices where remote cluster state is disabled, right? 

yes this understanding is also correct. When remote store is enabled but remote cluster state is not, the second check will pass(`shards with recovery source as remote are 0`) and you will  execute `routingTableBuilder.addAsRecovery(cursor);` which was happening earlier as well

Reply:
Then, why don't we just check if remote cluster state is enabled or not? Is it possible that remote cluster state is enabled but we will still have primary shards with recovery source not set to remote store?

Reply:
just checking remote cluster state would not be enough as even if remote cluster state is enabled we might not do a cluster state auto restore if any one of the previous cluster manager survived and was able to recover from local disk. then the cluster UUID would not change and you wont trigger auto restore.

in that case even if your remote cluster state is enabled, you still need to call `routingTableBuilder.addAsRecovery(cursor);`, as is the flow today.